src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -372,15 +400,15 @@ struct
     let implies ts (var, b) i =
       let tuple_cmp = Tuple2.eq (Option.eq ~eq:Int.equal) (Z.equal) in
       match var with
-      | None -> tuple_cmp (var, b) ts.(i)
-      | Some j -> tuple_cmp ts.(i) @@ Tuple2.map2 (Z.add b) ts.(j)
+      | None -> tuple_cmp (var, b) (EConj.get_rhs ts i)
+      | Some j -> tuple_cmp (EConj.get_rhs ts i) @@ Tuple2.map2 (Z.add b) (EConj.get_rhs ts j)
     in
     if env_comp = -2 || env_comp > 0 then false else
     if is_bot_env t1 || is_top t2 then true else
     if is_bot_env t2 || is_top t1 then false else
       let m1, m2 = Option.get t1.d, Option.get t2.d in
       let m1' = if env_comp = 0 then m1 else VarManagement.dim_add (Environment.dimchange t1.env t2.env) m1 in
-      GobArray.for_alli (fun i t -> implies m1' t i) m2
+      EConj.IntMap.fold (fun lhs rhs acc -> acc && implies m1' rhs lhs) (snd m2) true (* even on sparse m2, it suffices to check the non-trivial equalities, still present in sparse m2 *)
```

Comment:
`for_all` (https://ocaml.org/manual/5.1/api/Map.S.html)?

Reply:
Now the order of arguments is flipped, was that on purpose?

Reply:
`IntMap.for_all` wants `( int -> Rhs.t -> bool )`, and `(implies m1')` was  `(Rhs.t -> int -> bool)` for some reason, so I turned that around to fit together